### ISSUE TITLE
Firefox - Keyboard toggling of legend works only with Return, not with space bar

### ISSUE THREAD

**Issue by Toni-D1** [highcharts/highcharts#17839]
#### Expected behaviour
When legend entries are selected with keyboard navigation (using the a11y module) you should be able to toggle the series on and off using space bar or the return key.

#### Actual behaviour
When using Firefox browser only the return key works, the space bar does nothing. The space bar does work on the drop down menu in the following demo. 

#### Live demo with steps to reproduce
Can be reproduced on the basic template.
https://jsfiddle.net/highcharts/LLExL/

#### Product version
Highcharts 10.2.1

#### Affected browser(s)
Firefox 106.0b8

**Comment by oysteinmoseng** [comment]
This is likely due to this fix: https://github.com/highcharts/highcharts/pull/15585, which works around a firefox bug. We should check if this bug is still an issue, or if we can remove the Firefox-specific code now.
